app/services/merge_static_and_dynamic_targets.rb

```diff
@@ -0,0 +1,22 @@
+class MergeStaticAndDynamicTargets
+  # @param static_targets [Array<Static::Target>]
+  # @param report_targets [Array<Target>]
+  def initialize(static_targets, report_targets)
+    @static_targets = static_targets
+    @report_targets = report_targets
+  end
+
+  def call
+    @report_targets.map do |report_target|
+      match = @static_targets.detect do |static_target|
+        report_target.slug == static_target.slug
+      end
+      target_hash = match&.to_hash || {} # TODO: custom targets
+      target_hash[:year] = report_target.year
+      target_hash[:reported_percentage] = 0 # TODO:
```

Comment:
Not sure if the `reported_percentage` attribute is needed on targets under the `Planning` section.  
Beyond that everything is working fine :-)